### DISCUSSION TITLE
Outline of the Circle Does not seem to be rendered

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#2000] [Support Q&A] [answered]
I am using the below code to create a circle. Actually I do not want the fill. Just outline. But it does not seem to work. How can I draw a circle with just ouline? Fill is optional...

	osgEarth::Style circleStyle;
	circleStyle.getOrCreateSymbol<osgEarth::LineSymbol>()->stroke()->color() = osg::Vec4f(0, 1, 1, 1);
	circleStyle.getOrCreateSymbol<osgEarth::LineSymbol>()->stroke()->width() = 2.0f;
	circleStyle.getOrCreate<osgEarth::PolygonSymbol>()->fill()->color() =
		osgEarth::Color(osgEarth::Color::Cyan, 1.0); // transparent
	circleStyle.getOrCreate<osgEarth::AltitudeSymbol>()->clamping() = osgEarth::AltitudeSymbol::CLAMP_TO_TERRAIN;
	circleStyle.getOrCreate<AltitudeSymbol>()->technique() = AltitudeSymbol::TECHNIQUE_DRAPE;
	osgEarth::CircleNode* circle = new osgEarth::CircleNode;
		circle->set(
		point,
		Distance(300, Units::KILOMETERS), circleStyle);
	group->addChild(circle);

**Comment by nmielcarek** [comment, marked as answer]
This line:
```
circleStyle.getOrCreate<osgEarth::PolygonSymbol>()->fill()->color() =
	osgEarth::Color(osgEarth::Color::Cyan, 1.0); // transparent
```
1.0 = opaque. Set the last parameter to 0.0 to be transparent.

**Reply by nebsar** [reply]
It doesn't work. No outline is rendered. I do not know why.

**Reply by nmielcarek** [reply]
I modified the osgearth_annotation example to render the CircleNode around New Orleans in the style you posted and it appears to work fine.

![osgearth_circle](https://user-images.githubusercontent.com/4367015/156633384-7dad25a2-e456-4719-9855-2880af3fac61.png)

Does it work if you disable the clamping and technique lines? Maybe those functions are causing the lines to not show up properly on your system.

**Reply by nebsar** [reply]
ok I will try thanks.

**Reply by nebsar** [reply]
I tried the osgearth_annotation example. I did the same things. But no outlines. Which version of osgEarth do you use? I use the last version. Fill is rendered, but the outline is not rendered.

**Reply by nebsar** [reply]
Ok now it worked. But the strange thing is, when I load the earth file programmatically, circle is not rendered. When I use arguments it works. But I do not want to use arguments. Can you please try loading the earth file programmatically? Thanks

**Reply by nmielcarek** [reply]
When you say you are using "arguments", are you referring to loading a .earth file? I have tried both methods, in both versions of osgEarth (2.10 & 3.2) and they work fine. There must be something unique about your code or system that is preventing this.

Here is my system environment if you would like to compare:
```
osgearth_version --caps
[osgEarth]  [Registry] Note: GDAL_DATA environment variable is not set
[osgEarth]  [Capabilities] Capabilities: 
[osgEarth]  [Capabilities]   osgEarth Version:  3.2.0 build 128
[osgEarth]  [Capabilities]   OSG Version:       3.6.5
[osgEarth]  [Capabilities]   GDAL Version:      3.2.2
[osgEarth]  [Capabilities]   GEOS Version:      3.9.0
[osgEarth]  [Capabilities]   GPU Vendor:        NVIDIA Corporation
[osgEarth]  [Capabilities]   GPU Renderer:      NVIDIA GeForce GTX 1060 6GB/PCIe/SSE2
[osgEarth]  [Capabilities]   GL/Driver Version: 4.6.0 NVIDIA 495.46
[osgEarth]  [Capabilities]   GL Core Profile:   no
[osgEarth]  [Capabilities]   NVIDIA GL4:        no
[osgEarth]  [Capabilities]   NVIDIA GL4:        yes
```

**Reply by nebsar** [reply]
When I use the below code to load earth file, it does not work:
 osg::ref_ptr<osgViewer::Viewer> viewer = new osgViewer::Viewer;
 osg::ref_ptr<osg::Node> globe = osgDB::readNodeFile("resources/maps/earth_file/example.earth");

But when I use below code, accepting arguments, it works:
 osg::ref_ptr<osgViewer::Viewer> viewer = new osgViewer::Viewer(arguments);
 osg::Node* globe = MapNodeHelper().load(arguments, viewer);

I do not do any other changes.

**Reply by nmielcarek** [reply]
Add this after you create the viewer:
`GLUtils::setGlobalDefaults(viewer->getCamera()->getOrCreateStateSet());`

This FAQ item explains it:
https://docs.osgearth.org/en/latest/faq.html#why-are-my-lines-or-annotations-not-rendering

**Reply by nebsar** [reply]
Yes it did the trick.

**Comment by nebsar** [comment]
I really need help on this. A filled circle is rendered but no outline. Thanks
